mlflow/pyfunc/__init__.py

```diff
@@ -152,6 +293,244 @@ def _load_model_env(path, run_id=None):
     return _get_flavor_configuration(model_path=path, flavor_name=FLAVOR_NAME).get(ENV, None)
 
 
+def save_model(dst_path, loader_module=None, data_path=None, code_path=None, conda_env=None,
```

Comment:
Personally, I would prefer adding the new stuff as a separate flavor or PyFuncBuilder. Or move the older api to somewhere as a lower level api to construct pyfunc. I think it is really confusing to have one function with 2 conflicting sets of parameters.

Reply:
I also agree that this may be somewhat confusing. However, after discussing it with @mparkhe and @mateiz, we agree that this seems more pythonic than defining multiple `save_model` methods.